src/mlpack/methods/ann/layer/alpha_dropout.hpp

```diff
@@ -0,0 +1,183 @@
+/**
+ * @file alpha_dropout.hpp
+ * @author Dakshit Agrawal
+ *
+ * Definition of the Alpha-Dropout class, which implements a regularizer that
+ * randomly sets units to alpha-dash to prevent them from co-adapting and
+ * makes an affine transformation so as to keep the mean and variance of
+ * outputs at their original values.
+ *
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_LAYER_ALPHA_DROPOUT_HPP
+#define MLPACK_METHODS_ANN_LAYER_ALPHA_DROPOUT_HPP
+
+#include <mlpack/prereqs.hpp>
+
+// To use power function.
+#include <cmath>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The alpha - dropout layer is a regularizer that randomly with probability 'ratio'
+ * sets input values to alpha_dash.  An affine transformation is applied to the inputs.
+ * In the deterministic mode (during testing), the layer just gives the output.
```

Comment:
I think that we should clarify this documentation somewhat:

 * There's not clear documentation on what alpha_dash is or why that value is being chosen.  In the paper, it is chosen specifically to keep the outputs normalized.
 * There is no note that this is intended to be used with self-normalizing neural networks.
 * The meaning of `An affine transformation is applied to the inputs` is not clear.
 * I am not sure the comment `During training you should set deterministic to false` is necessary, since the user does not actually do that; the `FNN` and `RNN` classes do.